scripts/install_third_party.py

```diff
@@ -165,20 +190,76 @@ def return_json(source_url):
+    mandatory_keys = DOWNLOAD_FORMATS_TO_MANIFEST_KEYS[
+        dependency_type]['mandatory_keys']
+    optional_key_pairs= DOWNLOAD_FORMATS_TO_MANIFEST_KEYS[
+        dependency_type]['optional_key_pairs']
+    for key in mandatory_keys:
+        if key not in keys:
+            print '------------------------------------------'
+            print 'There is syntax error in this dependency'
+            print dependency_dict
+            print 'This key is missing or misspelled: "%s".' % key
+            print 'Exiting'
+            sys.exit()
+    if optional_key_pairs:
+        for optional_keys in optional_key_pairs:
+            if not any([True for key in optional_keys if key in keys]):
+                print '------------------------------------------'
+                print 'There is syntax error in this dependency'
+                print  dependency_dict
+                print 'This key is missing or misspelled: "%s".' % key
+                print 'Exiting'
+                sys.exit()
+    dependency_url = dependency_dict['url']
+    is_zip_file_format = dependency_type == _DOWNLOAD_FORMAT_ZIP
+    if (dependency_url.endswith('.zip') and not is_zip_file_format  or 
+        is_zip_file_format and not dependency_url.endswith('.zip')):
+            print '------------------------------------------'
+            print 'There is syntax error in this dependency'
+            print  dependency_dict
+            print 'This url  %s is invalid for %s file format.' % (
+                dependency_url, dependency_type)
+            print 'Exiting.'
+
+
+
+def validate_manifest(source_url):
+    """This validates syntax of the manifest.json
+    Args:
+      source_url: the url to the json file.
+    """
+    manifest_data = return_json(source_url)
+    dependencies = manifest_data['dependencies']
+    for data, dependency in dependencies.items():
+        for dependency_id,dependency_contents in dependency.items():
+            DOWNLOAD_FORMAT = dependency_contents['downloadFormat']
+            test_manifest_syntax(DOWNLOAD_FORMAT, dependency_contents)
+
+
 def download_manifest_files(source_url):
     """This download all files to the required folders
     Args:
-      source_url: the URL fof the json file.
+      source_url: the URL to the json file.
     """
+    validate_manifest(source_url)
     manifest_data = return_json(source_url)
     dependencies = manifest_data['dependencies']
-    for data in dependencies:
-        dependency = dependencies[data]
-        for dependency_id in dependency:
-            dependency_contents = dependency[dependency_id]
-            if 'srcUrl' in dependency_contents:
-                DEPENDENCY_REV = dependency_contents['version']
-                DEPENDENCY_URL = dependency_contents['srcUrl']
+    for data, dependency in dependencies.items():
+        for dependency_id,dependency_contents in dependency.items():
+            DEPENDENCY_REV = dependency_contents['version']
```

Comment:
ditto re non-capitalization, here and below. (But still capitalize _DOWNLOAD_FORMAT_FILES and the explicitly-declared constants at the top.)